src/engine/math_util.c

```diff
@@ -1340,7 +1340,7 @@ s32 ray_surface_intersect(Vec3f orig, Vec3f dir, f32 dir_length, struct Surface
     // Determine the cos(angle) difference between ray and surface normals.
     f32 det = vec3f_dot(e1, h);
     // Check if we're perpendicular from the surface.
-    if ((det > -NEAR_ZERO) && (det < NEAR_ZERO)) return FALSE;
+    if ((det < NEAR_ZERO)) return FALSE;
```

Comment:
too many parenthesis
```suggestion
    if (det < NEAR_ZERO) return FALSE;
```